### ISSUE TITLE
NamedPipeStreamReader might read in two messages as one

### ISSUE THREAD

**Issue by wilbaker** [Microsoft/VFSForGit#333]
Example functional test failure:

```
1) Failed : GVFS.FunctionalTests.Tests.EnlistmentPerTestCase.CaseOnlyFolderRenameTests.CaseRenameFoldersAndRemountAndRenameAgain()
Unmounting...Failed. Run 'gvfs log /GVFS.FT/test/3449aa1662704686a6a4' for more info.
Unrecognized response to unmount: AckComplete
  Expected: 0
  But was:  3
   at GVFS.Tests.Should.ValueShouldExtensions.ShouldEqual[T](T actualValue, T expectedValue, String message) in /Users/wilbaker/Repos/VFSForGit/src/GVFS/GVFS.Tests/Should/ValueShouldExtensions.cs:line 34
   at GVFS.FunctionalTests.Tools.GVFSProcess.CallGVFS(String args, Boolean failOnError) in /Users/wilbaker/Repos/VFSForGit/src/GVFS/GVFS.FunctionalTests/Tools/GVFSProcess.cs:line 109
   at GVFS.FunctionalTests.Tools.GVFSProcess.Unmount() in /Users/wilbaker/Repos/VFSForGit/src/GVFS/GVFS.FunctionalTests/Tools/GVFSProcess.cs:line 76
   at GVFS.FunctionalTests.Tools.GVFSFunctionalTestEnlistment.UnmountGVFS() in /Users/wilbaker/Repos/VFSForGit/src/GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs:line 227
   at GVFS.FunctionalTests.Tests.EnlistmentPerTestCase.CaseOnlyFolderRenameTests.CaseRenameFoldersAndRemountAndRenameAgain() in /Users/wilbaker/Repos/VFSForGit/src/GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/CaseOnlyFolderRenameTests.cs:line 56
```

I suspect this is due to the change to reading messages with `NamedPipeStreamReader.ReadMessage` instead of `ReadLine`.  If there are two messages in the stream and `ReadMessage` reads in both at once it looks like it will combine them into a single message.

**Comment by wilbaker** [comment]
@sanoursa @jrbriggs we might want to wait for this bug to be fixed before shipping the next release.

This change was introduced in #259

**Comment by jrbriggs** [comment]
Fixed in #343
